### ISSUE TITLE
Native dead letter queue support

### ISSUE THREAD

**Issue by sbochoa** [rebus-org/Rebus.AzureServiceBus#9]
Hi. I'm trying to implement a second level retry strategy based on scheduled retries but when the max retry count is reached I would like to send the message to the native ASB DLQ (can't do it directly as suggested [here ](https://stackoverflow.com/questions/43760922/how-to-forward-a-message-with-exceptions-to-rebus-error-queue/43769002#43769002) due to ASB implementation)  I couldn't find any documentation regarding ASB transport support for native DLQ, however I found [this ](https://stackoverflow.com/questions/35963019/what-happens-to-rebus-failed-messages#comment59583312_35963942) where @mookid8000 mention that it would be a good idea to disable Rebus' built-in deadlettering and pass that responsibility to ASB. Is there currently an option to do that? if not should I create another queue for errors? Currently the only way that I found to send the message to the native DLQ is by throwing exceptions until delivery count is reached but I would like to avoid this approach since it creates noise in our logs. Thanks!

**Comment by mookid8000** [comment]
> Is there currently an option to do that?

Unfortunately, no. 

> if not should I create another queue for errors?

Rebus automatically creates a queue named "error", but you can tell it to call it something else like this:

```csharp
Configure.With(...)
    .Options(b => b.SimpleRetryStrategy(errorQueueAddress: "alternative_error"))
    .(...)
```

I am not entirely sure that Rebus will use Azure Service Bus' built-in deadlettering in the future – it's just a thought.

Using an ordinary queue as the deadletter queue can be quite nice, because it makes it possible to set up another Rebus endpoint that consumes failed messages and does something with them.

Would the same thing be possible if ASB's built-in deadlettering was used?

What advantages would it bring?

**Comment by mookid8000** [comment]
> I could not find other discussions on this so could you please confirm that it is still currently not possible in Rebus?

It's still not (officially) possible, but if you e.g. set the number of delivery attempts on a queue lower than Rebus' configured max number of delivery attempts, the native deadlettering should kick in before Rebus gets to move the message to the error queue.

**Comment by Uleertel** [comment]
> > I could not find other discussions on this so could you please confirm that it is still currently not possible in Rebus?
> 
> It's still not (officially) possible, but if you e.g. set the number of delivery attempts on a queue lower than Rebus' configured max number of delivery attempts, the native deadlettering should kick in before Rebus gets to move the message to the error queue.

since then I managed to find the method UseNativeDeadlettering that seems to work as expected. Is is not a good workaround?

**Comment by Uleertel** [comment]
@mookid8000 since then I found UseNativeDeadlettering configuration extension and it seems to work as expected. Is it no a good solution?

**Comment by mookid8000** [comment]
Oh! Great! Well, sometimes it's a little bit hard to follow everything that happens across all of the Rebus.* repos, so I missed that one 😅 

I am sure it's a great solution, and it sounds like it's exactly what you're after 👍
